pkg/storage/unified/search/dashboard.go

```diff
@@ -0,0 +1,106 @@
+package search
+
+import (
+	"bytes"
+	"context"
+	"fmt"
+
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+
+	"github.com/grafana/grafana/pkg/apimachinery/utils"
+	"github.com/grafana/grafana/pkg/services/store/kind/dashboard"
+	"github.com/grafana/grafana/pkg/storage/unified/resource"
+)
+
+type DashboardDocument struct {
+	StandardDocumentFields `json:",inline"`
+
+	SchemaVersion   int64    `json:"schema_version,omitempty"`
+	LinkCount       int64    `json:"link_count,omitempty"`
+	PanelTypes      []string `json:"panel_type,omitempty"`
+	Transformations []string `json:"transformation,omitempty"`
+	DataSources     []string `json:"ds,omitempty"`
+	DataSourceTypes []string `json:"ds_type,omitempty"`
+
+	// Sortable stats
+	Stats map[string]int64 `json:"stats,omitempty"`
+}
+
+type DashboardDocumentBuilder struct {
+	// Scoped to a single tenant
+	Namespace string
+
+	// Cached stats for this namespace
+	// TODO, load this from apiserver request
+	Stats map[string]map[string]int64
```

Comment:
What kind of stats are expected to be put in here?

Reply:
Yes -- we need to have the stats from:
https://github.com/grafana/grafana-enterprise/tree/main/src/pkg/extensions/apiserver/registry/usageinsights

so we can support these sort fields:
<img width="834" alt="image" src="https://github.com/user-attachments/assets/2b4f9978-24ce-4d30-9b18-1270b8a52477">